conan/tools/microsoft/msbuild.py

```diff
@@ -27,10 +32,20 @@ def __init__(self, conanfile):
         if conanfile.settings.get_safe("os") == "WindowsCE":
             msvc_arch = conanfile.settings.get_safe("os.platform")
         self.platform = msvc_arch
+        self.toolset = msvs_toolset(conanfile)
 
     def command(self, sln, targets=None):
-        cmd = ('msbuild "%s" /p:Configuration=%s /p:Platform=%s'
-               % (sln, self.build_type, self.platform))
+        cmd = (f'msbuild "{sln}" /p:Configuration={self.build_type} '
+               f"/p:Platform={self.platform} /p:PlatformToolset={self.toolset}")
+
+        props_paths = []
+        for props_file in (MSBuildToolchain.filename, MSBuildDeps.filename):
+            props_path = os.path.join(self._conanfile.generators_folder, props_file)
+            if os.path.exists(props_path):
+                props_paths.append(props_path)
+        if props_paths:
```

Comment:
This cannot be made default, because that will be forcing the injection of whole dependencies into the current project, and there are many users that don't want that, but manual addition of individual dependencies into different subprojects.

At least, it should have an opt-out to allow users defining it.

Reply:
So keep this logic for `conantoolchain.props`, but allow to not autoconsume `conandeps.props`?

Regarding what users want, I'm wondering: users who manually edit their MSBuild project files, are they really using MSBuild conan helper? Or just MSBuildToolchain & MSBuildDeps?
This PR is really about providing the same high level logic in this toolchain/helper family than all other families in conan v2. It's the only family where the build helper does not automatically consume its toolchains. There is no opt-in or opt-out in Autotools helper to not globally inject AutotoolsDeps, so why a different behavior in MSBuild helper?

Reply:
From the feedback from users using this integration, it is very common that they have different subprojects and they want to inject different dependencies to different subprojects. If MSBuild pass ``conandeps.props`` by default, this cannot be done.

There are definitely other build systems that do the same, CMake uses the same pattern, users need to define ``find_package()`` and ``target_link_libraries()`` to the things that they want to use, it is not automagically applied to the whole project. For sure that is a built-in construct, but this level of control is what can be achieved with the MSBuild integration. The fact that it is not possiblel to do it in Autotools, for example, shouldn't be used as the design goal for it.

Reply:
I mostly agree for `*Deps` generators (even thought it's transparent in case of CMake), but `MSBuild` is the only build helper not consuming transparently its "glue toolchain".